app/components/CredentialStatusBadges/CredentialStatusBadges.tsx

```diff
@@ -17,10 +17,22 @@ function CredentialStatusBadges({
   const { styles, theme } = useDynamicStyles(dynamicStyleSheet);
   const checkPublicLink = useAsyncCallback<boolean>(hasPublicLink);
   const stablePublicRef = useRef<boolean | undefined>(undefined);
+  const currentCredentialIdRef = useRef<string | null>(null);
   const verifyCredential = precomputedVerification
     ? { loading: false, error: null, result: precomputedVerification }
     : useVerifyCredential(rawCredentialRecord);
 
+  // Get credential ID for comparison
+  const credentialId = (rawCredentialRecord as any)?._id?.toHexString?.() ?? (rawCredentialRecord as any)?._id;
```

Comment:
Extract this to its own function. Use [`credentialIdFor()`](https://github.com/openwallet-foundation-labs/learner-credential-wallet/blob/main/app/lib/decode.ts#L74), and replace its contents with this logic.